src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2730,10 +2730,15 @@ private ConstructorDeclarationSyntax ParseConstructorDeclaration(string typeName
                 }
 
                 BlockSyntax body;
+                ArrowExpressionClauseSyntax expressionBody;
                 SyntaxToken semicolon;
-                this.ParseBodyOrSemicolon(out body, out semicolon);
-
-                return _syntaxFactory.ConstructorDeclaration(attributes, modifiers.ToTokenList(), name, paramList, initializer, body, semicolon);
+                this.ParseBlockAndExpressionBodiesWithSemicolon(out body, out expressionBody, out semicolon);
+                if (expressionBody != null)
+                {
+                    expressionBody = CheckFeatureAvailability(expressionBody, MessageID.IDS_FeatureExpressionBodiedDeOrConstructor);
```

Comment:
ParseBlockAndExpressionBodiesWithSemicolon already checks for IDS_FeatureExpressionBodiedMethod feature. It  looks like it would be possible to get two errors for the same expression body under the "right" conditions. Consider passing feature ID to ParseBlockAndExpressionBodiesWithSemicolon as a parameter instead, it could have default value IDS_FeatureExpressionBodiedMethod.